S2VX.Game/Editor/ToolState/SelectToolState.cs

```diff
@@ -67,7 +125,7 @@ public class SelectToolState : S2VXToolState {
 
             foreach (var noteAndTime in SelectedNoteToTime) {
                 var note = noteAndTime.Key;
-                if (MouseIsOnNote(mousePos, Editor.NotesTimeline.NoteToTimelineNote[note])) {
+                if (Editor.NotesTimeline.NoteToTimelineNote.ContainsKey(note) && IsMouseOnTimelineNote(mousePos, Editor.NotesTimeline.NoteToTimelineNote[note])) {
```

Comment:
I dislike seeing long references like this used multiple times. It can contribute to excessive line length as well. In general, if you find yourself using long chains like this `Editor.NotesTimeline.NoteToTimelineNote`, I think it would easier on the eyes to just save a local variable `noteToTimelineNote` and reference that.